src/overlays/ovl_i2/fox_sx.c

```diff
@@ -1,81 +1,1986 @@
+                    gPlayer[0].state_1C8 = PLAYERSTATE_1C8_7;
+                    gPlayer[0].unk_1D0 = 0;
+                    gPlayer[0].unk_234 = 1;
+                }
+                Object_Kill(&boss->obj, &boss->sfxPos);
+                D_80177930 = boss->swork[6];
+            }
+            break;
+
+        default:
+            break;
+    }
+
+    D_i2_80195640 = 1;
+
+    Animation_DrawSkeleton(1, &D_6020C68, boss->vwork, func_i2_80192AF0, func_i2_80193208, boss, &gIdentityMatrix);
+
+    if (((boss->swork[1] != 0) && (boss->swork[3] > 0)) && ((boss->fwork[4] < 45.0f) || (boss->fwork[4] > 315.0f))) {
+        boss->swork[1]++;
+        if (boss->swork[1] >= 41) {
+            if (boss->swork[1] >= 51) {
+                boss->swork[1] = 0;
+            }
+            func_8007F11C(0x179, boss->fwork[20] + boss->obj.pos.x, boss->fwork[0x15] + boss->obj.pos.y,
+                          boss->fwork[0x16] + boss->obj.pos.z, 100.0f);
+            func_8007F11C(0x179, boss->fwork[0x17] + boss->obj.pos.x, boss->fwork[0x18] + boss->obj.pos.y,
+                          boss->fwork[0x19] + boss->obj.pos.z, 100.0f);
+        }
+    } else {
+        boss->swork[1] = 0;
+    }
+
+    if (boss->actionState != 35) {
+        boss->info.hitbox = SEGMENTED_TO_VIRTUAL(D_6032550);
+        boss->info.hitbox[43] = -211.0f + boss->fwork[16];
+        boss->info.hitbox[45] = -35.0f + boss->fwork[15];
+        boss->info.hitbox[47] = 442.0f + boss->fwork[14];
+        boss->info.hitbox[49] = -211.0f + boss->fwork[19];
+        boss->info.hitbox[51] = -35.0f + boss->fwork[18];
+        boss->info.hitbox[53] = -442.0f + boss->fwork[17];
+        boss->info.hitbox[55] = boss->fwork[10];
+        boss->info.hitbox[57] = boss->fwork[9];
+        boss->info.hitbox[59] = boss->fwork[8];
+        boss->info.hitbox[61] = boss->fwork[13];
+        boss->info.hitbox[63] = boss->fwork[12];
+        boss->info.hitbox[65] = boss->fwork[11];
+
+        if (attack) {
+            boss->info.hitbox[56] = 200.0f;
+            boss->info.hitbox[58] = 200.0f;
+            boss->info.hitbox[60] = 100.0f;
+            boss->info.hitbox[62] = 200.0f;
+            boss->info.hitbox[64] = 200.0f;
+            boss->info.hitbox[66] = 100.0f;
+        } else {
+            boss->info.hitbox[56] = 200.0f;
+            boss->info.hitbox[58] = 100.0f;
+            boss->info.hitbox[60] = 70.0f;
+            boss->info.hitbox[62] = 200.0f;
+            boss->info.hitbox[64] = 100.0f;
+            boss->info.hitbox[66] = 70.0f;
+        }
+    } else {
+        boss->info.hitbox = SEGMENTED_TO_VIRTUAL(D_6032768);
+    }
+
+    if (boss->swork[0] == 1) {
+        Math_SmoothStepToF(&boss->fwork[45], 1.0f, 1.0f, 0.05f, 0);
+        return;
+    }
+
+    Math_SmoothStepToF(&boss->fwork[45], 0.0f, 1.0f, 0.05f, 0);
+}
+
+s32 func_i2_80192AF0(s32 arg0, Gfx** arg1, Vec3f* arg2, Vec3f* arg3, void* arg4) {
+    Vec3f sp64 = D_i2_80195674;
```

Comment:
```suggestion
    Vec3f sp64 = {0.0f, 0.0f, 0.0f};
```
This sort of thing is basically always a stack variable intializer. You'll have to move the data around to accommodate this.